src/openms/include/OpenMS/TRANSFORMATIONS/FEATUREFINDER/ExtendedIsotopeFitter1D.h

```diff
@@ -73,7 +73,7 @@ namespace OpenMS
     }
 
     /// return interpolation model
-    QualityType fit1d(const RawDataArrayType & range, InterpolationModel * & model) override;
+    QualityType fit1d(const RawDataArrayType & range, std::unique_ptr<InterpolationModel>& model) override;
```

Comment:
can we wrap that in pyOpenMS?

Reply:
and would it make sense at all because of polymorphism ???

Reply:
Of course, you can. But not with autowrap.

Reply:
Easy workaround is to offer a function that accepts a shared pointer since autowrap has a conversion provider for this (essentially passing the "inst" member of the already wrapped python class).

Reply:
or just pass a smart_ptr ?

Reply:
introducing a shared_ptr where a unique_ptr would suffice is bad design.
is there a possibility to wrap unique_ptr? this would be cleaner.